pennylane/ops/op_math/sprod.py

```diff
@@ -244,7 +244,11 @@ def sparse_matrix(self, wire_order=None):
         Returns:
             :class:`scipy.sparse._csr.csr_matrix`: sparse matrix representation
         """
-        return self.base.sparse_matrix(wire_order=wire_order).multiply(self.scalar)
+        if self._pauli_rep:  # Get the sparse matrix from the PauliSentence representation
+            return self._pauli_rep.to_mat(wire_order=wire_order or self.wires, format="csr")
+        mat = self.base.sparse_matrix(wire_order=wire_order).multiply(self.scalar)
+        mat.eliminate_zeros()
```

Comment:
Is this in case `self.scalar` is `0.0` or doesn't `sparse_matrix` already calls `eliminate_zeros`?

Reply:
Currently, if `self.scalar` is 0, we'd get a data array of 0s unless we eliminate them explicitly:
```python
>>> qml.sprod(0, qml.PauliX(0)).sparse_matrix().data
array([0, 0])
```